src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/CleanProperties.groovy

```diff
@@ -1,32 +1,35 @@
-/*
- * Source https://stackoverflow.com/a/39043903/519333
- */
 package com.github.jengelman.gradle.plugins.shadow.internal
 
 class CleanProperties extends Properties {
-    private static class StripFirstLineStream extends FilterOutputStream {
 
-        private boolean firstLineSeen = false
+    private static class StripCommentsWithTimestampBufferedWriter extends BufferedWriter {
 
-        StripFirstLineStream(final OutputStream out) {
-            super(out)
+        private final int lengthOfExpectedTimestamp
+
+        StripCommentsWithTimestampBufferedWriter(final Writer out) {
+            super(out);
+
+            lengthOfExpectedTimestamp = ("#" + new Date().toString()).length()
```

Comment:
This `Date` is related to your locale, how can we make sure the device locale is matched with the jar locale?

Reply:
The idea of this is to just remove the comment lines which include the timestamp, which is generated when the properties file is written, so I assume locale is the same.  Timestamp is not coming from other jar files.  Either way I agree is a bit tricky.

Another option could be try to parse the date to see if it's a date or not, or maybe keep it simple and strip all comment lines.  

Probably these changes are a mitigation as the real solution should come from the component writing the properties file, but anyway is more solid than the previous one which was removing only the first line.